File: Java/3.longestSubstringWithoutRepeatingCharacters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s == null || s.length() == 0) {
            return 0;
        }
        int[] hash = new int[256];
        int begin = 0, end = 0;
        int repeat = 0, res = 0;
        while (end < s.length()) {
            if (hash[s.charAt(end++)]++ > 0) {
                repeat++;
            }
            while (repeat > 0) {
                if (hash[s.charAt(begin++)]-- > 1) {
                    repeat--;
                }
            }
            res = Math.max(res, end - begin);
        }
        return res;
    }
}

File: Javascript/3.longestSubstringWithoutRepeatingCharacters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let res = 0;
    for (let i = 0; i < s.length; i++) {
        let set = new Set();
        set.add(s.charAt(i));
        for (let j = i + 1; j < s.length; j++) {
            if (set.has(s.charAt(j))) {
                break;
            }
            set.add(s.charAt(j));
        }
        res = Math.max(res, set.size);
    }
    return res;
};